File: workspace/2023/2023-01-17/fibonacci-number.c/fibonacci-number.c
#include <stdio.h>

void fibonacci(int n);

int main(void)
{
    fibonacci(20);

    return 0;
}

void fibonacci(int n)
{
    int i;
    int a = 0;
    int b = 1;

    for (i = 0; i <= n; i++)
    {
        printf("%d\n", a);
        b = a + b;
        a = b - a;
    }
}

File: workspace/2023/2023-03-13/fibonacci-number.forth/fibonacci-number.fs

: _fibonacci recursive ( a b n -- fn )
    dup 0 = if
        drop drop . ( a b n -- )
    else
        1 -         ( a b n -- a b n-1 )
        rot rot     ( a b n-1 -- n-1 a b )
        over .
        tuck +      ( n-1 a b -- n-1 b a+b )
        rot         ( n-1 b a+b -- b a+b n-1 )
        _fibonacci  ( b a+b n-1 -- )
    endif
;

: fibonacci ( n -- fn )
    0 1 rot     ( n -- 0 1 n )
    _fibonacci  ( 0 1 n -- fn )
;

20 fibonacci

